### PR TITLE
Removes apps from project specs and speeds up the query

### LINKED ISSUE #2073: Painfully slow `odo project list`
`oc get projects` takes under the second, but `odo project list` is 13 seconds :-(

```
▶ gtime -f  "%es" oc get projects
NAME                                                    DISPLAY NAME   STATUS
default                                                                Active
kube-node-lease                                                        Active
kube-public                                                            Active
kube-system                                                            Active
oibwrltsgf                                                             Active
openshift                                                              Active
openshift-apiserver                                                    Active
openshift-apiserver-operator                                           Active
openshift-authentication                                               Active
openshift-authentication-operator                                      Active
openshift-cloud-credential-operator                                    Active
openshift-cluster-machine-approver                                     Active
openshift-cluster-node-tuning-operator                                 Active
openshift-cluster-samples-operator                                     Active
openshift-cluster-storage-operator                                     Active
openshift-cluster-version                                              Active
openshift-config                                                       Active
openshift-config-managed                                               Active
openshift-console                                                      Active
openshift-console-operator                                             Active
openshift-controller-manager                                           Active
openshift-controller-manager-operator                                  Active
openshift-dns                                                          Active
openshift-dns-operator                                                 Active
openshift-etcd                                                         Active
openshift-image-registry                                               Active
openshift-infra                                                        Active
openshift-ingress                                                      Active
openshift-ingress-operator                                             Active
openshift-insights                                                     Active
openshift-kube-apiserver                                               Active
openshift-kube-apiserver-operator                                      Active
openshift-kube-controller-manager                                      Active
openshift-kube-controller-manager-operator                             Active
openshift-kube-scheduler                                               Active
openshift-kube-scheduler-operator                                      Active
openshift-machine-api                                                  Active
openshift-machine-config-operator                                      Active
openshift-marketplace                                                  Active
openshift-monitoring                                                   Active
openshift-multus                                                       Active
openshift-network-operator                                             Active
openshift-node                                                         Active
openshift-operator-lifecycle-manager                                   Active
openshift-operators                                                    Active
openshift-sdn                                                          Active
openshift-service-ca                                                   Active
openshift-service-ca-operator                                          Active
openshift-service-catalog-apiserver-operator                           Active
openshift-service-catalog-controller-manager-operator                  Active
rgzcsmmdch                                                             Active
uhrnluzzwm                                                             Active
0.73s

▶ gtime -f  "%es" odo project list
ACTIVE     NAME
           default
           kube-node-lease
           kube-public
           kube-system
           oibwrltsgf
           openshift
           openshift-apiserver
           openshift-apiserver-operator
           openshift-authentication
           openshift-authentication-operator
           openshift-cloud-credential-operator
           openshift-cluster-machine-approver
           openshift-cluster-node-tuning-operator
           openshift-cluster-samples-operator
           openshift-cluster-storage-operator
           openshift-cluster-version
           openshift-config
           openshift-config-managed
           openshift-console
           openshift-console-operator
           openshift-controller-manager
           openshift-controller-manager-operator
           openshift-dns
           openshift-dns-operator
           openshift-etcd
           openshift-image-registry
           openshift-infra
           openshift-ingress
           openshift-ingress-operator
           openshift-insights
           openshift-kube-apiserver
           openshift-kube-apiserver-operator
           openshift-kube-controller-manager
           openshift-kube-controller-manager-operator
           openshift-kube-scheduler
           openshift-kube-scheduler-operator
           openshift-machine-api
           openshift-machine-config-operator
           openshift-marketplace
           openshift-monitoring
           openshift-multus
           openshift-network-operator
           openshift-node
           openshift-operator-lifecycle-manager
           openshift-operators
           openshift-sdn
           openshift-service-ca
           openshift-service-ca-operator
           openshift-service-catalog-apiserver-operator
           openshift-service-catalog-controller-manager-operator
           rgzcsmmdch
           uhrnluzzwm
13.62s
```

### PR DESCRIPTION
**Pull request by mik-dass**
fixes #2073 

How to test:
1. Execute `project list` and `project list -o json` and compare speeds with the master branch

### CODE REVIEW

File: pkg/project/project.go [hunk 1]
```diff
@@ -50,7 +49,7 @@ func Delete(client *occlient.Client, projectName string) error {
 	return nil
 }
 
-func DescribeProjects(client *occlient.Client) (ProjectList, error) {
+func List(client *occlient.Client) (ProjectList, error) {
```
Comment (dharmit) [inline pkg/project/project.go]: Can we please add comments for the function?

**Comment by dharmit** [comment]
@mik-dass the output of `odo project list -o json` looks similar for both `master` and this PR. But earlier we used to grab info about applications in the projects as well, right? We didn't use that info about applications anywhere? :thinking:

**Comment by mik-dass** [comment]
> We didn't use that info about applications anywhere?

Since the names of the apps belonging to the project was not used by the vscode adapter team. we decided to remove it from the spec itself. They use `app list` instead

**Comment by kadel** [comment]
It looks like that somethings is failing 

```
• Failure [6.297 seconds]
odo project command tests
/go/src/github.com/openshift/odo/tests/integration/cmd_project_test.go:13
  when running project command app parameter in directory that doesn't contain .odo config directory
  /go/src/github.com/openshift/odo/tests/integration/cmd_project_test.go:41
    should successfully execute list along with machine readable output [It]
    /go/src/github.com/openshift/odo/tests/integration/cmd_project_test.go:42
    Expected
      
        <string>: ACTIVE     NAME
                   myproject
                   ofoxzlehss
                   typareevak
        
    to contain substring
        <string>: zjgsrmgbtj
    /go/src/github.com/openshift/odo/tests/integration/cmd_project_test.go:44
```

**Comment by mik-dass** [comment]
> It looks like that somethings is failing

It looks like the project created is not fully ready before list is performed

File: Makefile [hunk 2]
```diff
@@ -133,6 +133,12 @@ test-cmd-service:
 	ginkgo -v -nodes=$(TEST_EXEC_NODES) -focus="odo service command tests" \
 	slowSpecThreshold=$(SLOW_SPEC_THRESHOLD) -randomizeAllSpecs  tests/integration/ -timeout $(TIMEOUT)
 
+# Run odo app command tests
```
Comment (amitkrout) [inline Makefile]: change to ```Run odo project command tests```

File: tests/integration/cmd_project_test.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	"os"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/odo/tests/helper"
+)
+
+var _ = Describe("odo project command tests", func() {
```
Comment (amitkrout) [inline tests/integration/cmd_project_test.go]: +100 for new test file. Later it will be easy to improve with all possible project command scenarios.

**Comment by amitkrout** [comment]
> > It looks like that somethings is failing
> 
> It looks like the project created is not fully ready before list is performed even when `-w` flag is used

@mrinal Is it reproducible locally ?

**Comment by mik-dass** [comment]
> @mrinal Is it reproducible locally ?

No. All project tests pass locally. I have added sleep for debugging purpose.

**Comment by mik-dass** [comment]
https://travis-ci.com/openshift/odo/jobs/233964101#L2793

```
Running odo with args [odo project list]
[odo] ACTIVE     NAME
[odo] *          kdiadkbatt
[odo]            myproject
Running odo with args [odo project list]
[odo] ACTIVE     NAME
[odo]            kbcwtnzrez
[odo] *          kdiadkbatt
[odo]            myproject
```
@kadel @amitkrout after applying sleep I can confirm, the wait flag is not working properly

File: tests/integration/cmd_project_test.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+		It("should display the help", func() {
+			appHelp := helper.CmdShouldPass("odo", "project", "-h")
+			Expect(appHelp).To(ContainSubstring("Perform project operations"))
+		})
+	})
+
+	Context("when running project command app parameter in directory that doesn't contain .odo config directory", func() {
+		It("should successfully execute list along with machine readable output", func() {
+			helper.CmdRunner("odo", "project", "list")
+			time.Sleep(1 * time.Second)
```
Comment (amitkrout) [inline tests/integration/cmd_project_test.go]: @mik-dass you missed it to remove i guess

Comment (mik-dass) [reply tests/integration/cmd_project_test.go]: Because of #2117, we need to either comment out the test or fix the issue first

**Comment by mik-dass** [comment]
> @kadel @amitkrout after applying sleep I can confirm, the wait flag is not working properly. Created issue #2117

@amitkrout @kadel Should I comment out the test for now?

File: tests/integration/cmd_project_test.go (hunk 4 above)
Comment (kadel) [reply tests/integration/cmd_project_test.go]: I don't like this. We should never fix issues with sleep. But I guess it is kind of temporary ok :-( as long as we track it in issue

Comment (mik-dass) [reply tests/integration/cmd_project_test.go]: I want to remove this and comment out the test for now or should we keep this?

File: pkg/project/project.go (hunk 1 above)
Comment (girishramnani) [reply pkg/project/project.go]: I can see comments now so assuming this as resolved

**Comment by girishramnani** [comment]
this PR changes the out of `odo project list`. probably a good idea to mark it as breaking change and add in the change log.
@kadel @mik-dass

**Comment by mik-dass** [comment]
> this PR changes the out of odo project list. probably a good idea to mark it as breaking change and add in the change log.

:+1:

**Pull request merged**